# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../nas/gcc4.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -Nur nas-1.7.orig/clients/audio/auconvert/auconvert.c nas-1.7/clients/audio/auconvert/auconvert.c
--- nas-1.7.orig/clients/audio/auconvert/auconvert.c	2000-12-13 06:22:31.000000000 +0200
+++ nas-1.7/clients/audio/auconvert/auconvert.c	2005-10-23 20:16:12.000000000 +0300
@@ -383,7 +383,7 @@
 	    fatalError("Error writing output file");
     }
     else
-	rateConvert(in, out, numBytes, data);
+	rateConvert(in, out, numBytes, (char *)data);
 
     if (SoundCloseFile(out))
 	fatalError("Error closing output file");
diff -Nur nas-1.7.orig/clients/audio/auinfo/auinfo.c nas-1.7/clients/audio/auinfo/auinfo.c
--- nas-1.7.orig/clients/audio/auinfo/auinfo.c	1999-09-05 04:42:55.000000000 +0200
+++ nas-1.7/clients/audio/auinfo/auinfo.c	2005-10-23 20:16:12.000000000 +0300
@@ -26,6 +26,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <audio/audiolib.h>
 
diff -Nur nas-1.7.orig/clients/audio/aupanel/aupanel.c nas-1.7/clients/audio/aupanel/aupanel.c
--- nas-1.7.orig/clients/audio/aupanel/aupanel.c	2002-07-21 03:13:30.000000000 +0200
+++ nas-1.7/clients/audio/aupanel/aupanel.c	2005-10-23 20:16:12.000000000 +0300
@@ -28,6 +28,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #ifndef SYSV
 #include <audio/Aos.h>		/* for string and other os stuff */
 #endif
diff -Nur nas-1.7.orig/clients/audio/auplay/auplay.c nas-1.7/clients/audio/auplay/auplay.c
--- nas-1.7.orig/clients/audio/auplay/auplay.c	2001-03-18 22:36:38.000000000 +0200
+++ nas-1.7/clients/audio/auplay/auplay.c	2005-10-23 20:16:12.000000000 +0300
@@ -30,7 +30,9 @@
 #include	<unistd.h>
 #include	<fcntl.h>
 #include	<stdio.h>
+#include    <stdlib.h>
 #include	<signal.h>
+#include    <string.h>
 #include	<audio/audiolib.h>
 #include	<audio/soundlib.h>
 
diff -Nur nas-1.7.orig/clients/audio/aurecord/aurecord.c nas-1.7/clients/audio/aurecord/aurecord.c
--- nas-1.7.orig/clients/audio/aurecord/aurecord.c	1999-05-08 20:47:18.000000000 +0200
+++ nas-1.7/clients/audio/aurecord/aurecord.c	2005-10-23 20:16:12.000000000 +0300
@@ -28,6 +28,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <signal.h>
 #include <audio/Aos.h>			/* for string and other os stuff */
 #include <audio/Afuncs.h> 		/* for bcopy et. al. */
diff -Nur nas-1.7.orig/clients/audio/auwave/auwave.c nas-1.7/clients/audio/auwave/auwave.c
--- nas-1.7.orig/clients/audio/auwave/auwave.c	2002-07-21 03:13:30.000000000 +0200
+++ nas-1.7/clients/audio/auwave/auwave.c	2005-10-23 20:16:12.000000000 +0300
@@ -28,6 +28,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #ifndef SYSV
 #include <audio/Aos.h> 		/* for string and other os stuff */
 #endif
diff -Nur nas-1.7.orig/clients/audio/examples/bundle.c nas-1.7/clients/audio/examples/bundle.c
--- nas-1.7.orig/clients/audio/examples/bundle.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/examples/bundle.c	2005-10-23 20:16:12.000000000 +0300
@@ -17,6 +17,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <audio/audiolib.h>
 #include <audio/soundlib.h>
 
diff -Nur nas-1.7.orig/clients/audio/examples/loopBucket.c nas-1.7/clients/audio/examples/loopBucket.c
--- nas-1.7.orig/clients/audio/examples/loopBucket.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/examples/loopBucket.c	2005-10-23 20:16:12.000000000 +0300
@@ -10,6 +10,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <audio/audiolib.h>
 #include <audio/soundlib.h>
 
diff -Nur nas-1.7.orig/clients/audio/examples/playFile.c nas-1.7/clients/audio/examples/playFile.c
--- nas-1.7.orig/clients/audio/examples/playFile.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/examples/playFile.c	2005-10-23 20:16:12.000000000 +0300
@@ -9,6 +9,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <audio/audiolib.h>
 #include <audio/soundlib.h>
 
diff -Nur nas-1.7.orig/clients/audio/examples/playFromBckt.c nas-1.7/clients/audio/examples/playFromBckt.c
--- nas-1.7.orig/clients/audio/examples/playFromBckt.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/examples/playFromBckt.c	2005-10-23 20:16:12.000000000 +0300
@@ -9,6 +9,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <audio/audiolib.h>
 #include <audio/soundlib.h>
 
diff -Nur nas-1.7.orig/clients/audio/examples/playSimul.c nas-1.7/clients/audio/examples/playSimul.c
--- nas-1.7.orig/clients/audio/examples/playSimul.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/examples/playSimul.c	2005-10-23 20:16:12.000000000 +0300
@@ -9,6 +9,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <audio/audiolib.h>
 #include <audio/soundlib.h>
 
@@ -48,7 +49,7 @@
 	    {
 		arg++;
 		argc--;
-		volume = atoi(arg);
+		volume = atoi(*arg);
 	    }
 	}
 	else if (AuSoundPlayFromFile(aud, *arg, AuNone,
diff -Nur nas-1.7.orig/clients/audio/examples/plyBcktAsync.c nas-1.7/clients/audio/examples/plyBcktAsync.c
--- nas-1.7.orig/clients/audio/examples/plyBcktAsync.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/examples/plyBcktAsync.c	2005-10-23 20:16:12.000000000 +0300
@@ -7,6 +7,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <audio/audiolib.h>
 #include <audio/soundlib.h>
 
diff -Nur nas-1.7.orig/clients/audio/examples/recordBucket.c nas-1.7/clients/audio/examples/recordBucket.c
--- nas-1.7.orig/clients/audio/examples/recordBucket.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/examples/recordBucket.c	2005-10-23 20:16:12.000000000 +0300
@@ -11,6 +11,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <audio/audiolib.h>
 #include <audio/soundlib.h>
 
diff -Nur nas-1.7.orig/clients/audio/util/playbucket.c nas-1.7/clients/audio/util/playbucket.c
--- nas-1.7.orig/clients/audio/util/playbucket.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/util/playbucket.c	2005-10-23 20:16:12.000000000 +0300
@@ -23,6 +23,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include <audio/audiolib.h>
 #include <audio/soundlib.h>
diff -Nur nas-1.7.orig/clients/audio/widgets/Slider.c nas-1.7/clients/audio/widgets/Slider.c
--- nas-1.7.orig/clients/audio/widgets/Slider.c	1999-05-08 20:47:19.000000000 +0200
+++ nas-1.7/clients/audio/widgets/Slider.c	2005-10-23 20:16:12.000000000 +0300
@@ -31,6 +31,7 @@
 #include <X11/StringDefs.h>
 #include <X11/Xaw/Label.h>
 #include <X11/Xaw/Scrollbar.h>
+#include <stdio.h>
 #include "SliderP.h"
 
 static XtResource resources[] = {
diff -Nur nas-1.7.orig/server/dia/auprocess.c nas-1.7/server/dia/auprocess.c
--- nas-1.7.orig/server/dia/auprocess.c	2004-06-21 00:31:55.000000000 +0300
+++ nas-1.7/server/dia/auprocess.c	2005-10-23 20:16:12.000000000 +0300
@@ -25,6 +25,7 @@
 #define _AUPROCESS_C_
 #define OUTPUT_RANGE_CHECK
 
+#include <string.h>
 #include <audio/audio.h>
 #include <audio/Aproto.h>
 #include "au.h"
diff -Nur nas-1.7.orig/server/dia/auutil.c nas-1.7/server/dia/auutil.c
--- nas-1.7.orig/server/dia/auutil.c	1999-09-05 04:43:13.000000000 +0200
+++ nas-1.7/server/dia/auutil.c	2005-10-23 20:16:12.000000000 +0300
@@ -173,7 +173,7 @@
 AuCreateResourceTypes()
 {
     void            AuFreeComponent();
-    static void     freeFlow();
+    void     freeFlow();
 
     auComponentType = CreateNewResourceType(AuFreeComponent);
     auFlowType = CreateNewResourceType(freeFlow);
diff -Nur nas-1.7.orig/server/dia/main.c nas-1.7/server/dia/main.c
--- nas-1.7.orig/server/dia/main.c	2004-09-11 22:23:33.000000000 +0300
+++ nas-1.7/server/dia/main.c	2005-10-23 20:16:12.000000000 +0300
@@ -52,6 +52,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <audio/audio.h>
 #include <audio/Aproto.h>
 #include "NasConfig.h"
diff -Nur nas-1.7.orig/server/os/access.c nas-1.7/server/os/access.c
--- nas-1.7.orig/server/os/access.c	1999-07-08 02:51:49.000000000 +0200
+++ nas-1.7/server/os/access.c	2005-10-23 20:16:12.000000000 +0300
@@ -583,7 +583,8 @@
 AuthorizedClient(client)
     ClientPtr client;
 {
-    int    		alen, family;
+    int    		family;
+	socklen_t   alen;
 #ifndef AMOEBA
     struct sockaddr	from;
 #else
diff -Nur nas-1.7.orig/server/os/connection.c nas-1.7/server/os/connection.c
--- nas-1.7.orig/server/os/connection.c	2004-02-15 01:12:22.000000000 +0200
+++ nas-1.7/server/os/connection.c	2005-10-23 20:16:12.000000000 +0300
@@ -1402,7 +1402,7 @@
 	struct sockaddr_dn dn;
 #endif /* DNETCONN */
     } from;
-    int	fromlen = sizeof (from);
+    socklen_t	fromlen = sizeof (from);
     AuID	 auth_id;
 
     auth_id = CheckAuthorization (proto_n, auth_proto,
@@ -1480,7 +1480,7 @@
 	struct sockaddr_dn dn;
 #endif /* DNETCONN */
     } from;
-    int	fromlen;
+    socklen_t	fromlen;
 #endif /* TCP_NODELAY */
 
     readyconnections = (((long)closure) & WellKnownConnections);
@@ -1504,7 +1504,7 @@
 	readyconnections &= ~(1 << curconn);
 	if ((newconn = accept (curconn,
 			      (struct sockaddr *) NULL, 
-			      (int *)NULL)) < 0) 
+			      (socklen_t *)NULL)) < 0) 
 	    continue;
 	if (newconn > lastfdesc)
 	{
diff -Nur nas-1.7.orig/server/os/utils.c nas-1.7/server/os/utils.c
--- nas-1.7.orig/server/os/utils.c	2004-09-11 22:23:33.000000000 +0300
+++ nas-1.7/server/os/utils.c	2005-10-23 20:16:12.000000000 +0300
@@ -51,6 +51,7 @@
 #include <audio/audio.h>
 #include <audio/Aos.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include "nasconf.h"
 #include "misc.h"
 #include "opaque.h"
